net/browserless_agent/src/browserless_agent.ts

```diff
@@ -0,0 +1,254 @@
+        error: any;
+      };
+    };
+
+export const browserlessAgent: AgentFunction<BrowserlessParams, BrowserlessResult, BrowserlessInputs, DefaultConfigData> = async ({ namedInputs, params }) => {
+  const { url, operation, options, headers, cookies } = namedInputs;
+  const throwError = params?.throwError ?? false;
+
+  const browserlessEndpoint = "https://chrome.browserless.io";
+  const browserlessToken = process.env.BROWSERLESS_API_TOKEN;
+
+  // Check if API token is provided
+  if (!browserlessToken) {
+    const errorMessage = "Browserless API token is required. Please set the BROWSERLESS_API_TOKEN environment variable.";
+    throw new Error(errorMessage);
+  }
+
+  // Build the endpoint with token
+  const endpoint = `${browserlessEndpoint}?token=${browserlessToken}`;
+
+  // Select appropriate endpoint based on operation type
+  const operationType = operation || "content";
+  let apiPath: string;
+
+  switch (operationType) {
+  case "screenshot":
+    apiPath = "/screenshot";
+    break;
+  case "content":
+  default:
+    apiPath = "/content";
+    break;
+  }
+
+  const fullEndpoint = `${endpoint}${apiPath}`;
+
+  // Build request body
+  const requestBody: Record<string, any> = {
+    url,
+    ...options,
+  };
+
+  if (cookies && Array.isArray(cookies) && cookies.length > 0) {
+    requestBody.cookies = cookies;
+  }
+
+  // Return request information in debug mode
+  if (params?.debug) {
+    return {
+      url: fullEndpoint,
+      method: "POST",
+      headers: headers || { "Content-Type": "application/json" },
+      body: requestBody,
+    } as BrowserlessResult;
+  }
+
+  // API request options
+  const fetchOptions: RequestInit = {
+    method: "POST",
+    headers: new Headers({
+      "Content-Type": "application/json",
+      ...(headers || {}),
+    }),
+    body: JSON.stringify(requestBody),
+  };
+
+  try {
+    const response = await fetch(fullEndpoint, fetchOptions);
+
+    if (!response.ok) {
+      const status = response.status;
+      const error = await response.text();
+
+      if (throwError) {
+        throw new Error(`Browserless HTTP error: ${status}`);
+      }
+
+      return {
+        onError: {
+          message: `Browserless HTTP error: ${status}`,
+          status,
+          error,
+        },
+      };
+    }
+
+    // Process result based on response type
+    const responseType = params?.type || (operationType === "screenshot" ? "buffer" : "text");
+
+    switch (responseType) {
+    case "json":
+      return (await response.json()) as BrowserlessResult;
+    case "text":
+      return (await response.text()) as BrowserlessResult;
+    case "buffer":
+      return {
+        base64: Buffer.from(await response.arrayBuffer()).toString("base64"),
+        contentType: response.headers.get("Content-Type") || "application/octet-stream",
+      } as BufferResponse;
+    default:
+      return (await response.text()) as BrowserlessResult;
+    }
+  } catch (error: any) {
```

Comment:
It's better to use TypeScript infer the type and handle errors accordingly, rather than using any.